dll/win32/shdocvw/CFavBand.cpp

```diff
@@ -0,0 +1,648 @@
+    {
+        //SetFocus();
+        SetActiveWindow();
+    }
+
+    if (lpMsg)
+    {
+        TranslateMessage(lpMsg);
+        DispatchMessage(lpMsg);
+    }
+
+    return S_OK;
+}
+
+STDMETHODIMP CFavBand::HasFocusIO()
+{
+    return m_bFocused ? S_OK : S_FALSE;
+}
+
+STDMETHODIMP CFavBand::TranslateAcceleratorIO(LPMSG lpMsg)
+{
+    if (lpMsg->hwnd == m_hWnd)
+    {
+        TranslateMessage(lpMsg);
+        DispatchMessage(lpMsg);
+        return S_OK;
+    }
+
+    return S_FALSE;
+}
+
+// *** IPersist ***
+
+STDMETHODIMP CFavBand::GetClassID(CLSID *pClassID)
+{
+    if (!pClassID)
+        return E_POINTER;
+    memcpy(pClassID, &CLSID_SH_FavBand, sizeof(CLSID));
+    return S_OK;
+}
+
+
+// *** IPersistStream ***
+
+STDMETHODIMP CFavBand::IsDirty()
+{
+    UNIMPLEMENTED;
+    return E_NOTIMPL;
+}
+
+STDMETHODIMP CFavBand::Load(IStream *pStm)
+{
+    UNIMPLEMENTED;
+    return E_NOTIMPL;
+}
+
+STDMETHODIMP CFavBand::Save(IStream *pStm, BOOL fClearDirty)
+{
+    UNIMPLEMENTED;
+    return E_NOTIMPL;
+}
+
+STDMETHODIMP CFavBand::GetSizeMax(ULARGE_INTEGER *pcbSize)
+{
+    // TODO: calculate max size
+    UNIMPLEMENTED;
+    return E_NOTIMPL;
+}
+
+// *** IWinEventHandler ***
+
+STDMETHODIMP CFavBand::OnWinEvent(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam, LRESULT *theResult)
+{
+    UNIMPLEMENTED;
+    return E_NOTIMPL;
+}
+
+STDMETHODIMP CFavBand::IsWindowOwner(HWND hWnd)
+{
+    return (hWnd == m_hWnd) ? S_OK : S_FALSE;
+}
+
+// *** IBandNavigate ***
+
+STDMETHODIMP CFavBand::Select(long paramC)
+{
+    UNIMPLEMENTED;
+    return E_NOTIMPL;
+}
+
+// *** INamespaceProxy ***
+
+STDMETHODIMP CFavBand::GetNavigateTarget(long paramC, long param10, long param14)
+{
+    UNIMPLEMENTED;
+    return E_NOTIMPL;
+}
+
+STDMETHODIMP CFavBand::Invoke(long paramC)
+{
+    UNIMPLEMENTED;
+    return E_NOTIMPL;
+}
+
+STDMETHODIMP CFavBand::OnSelectionChanged(long paramC)
+{
+    UNIMPLEMENTED;
+    return E_NOTIMPL;
+}
+
+STDMETHODIMP CFavBand::RefreshFlags(long paramC, long param10, long param14)
+{
+    UNIMPLEMENTED;
+    return E_NOTIMPL;
+}
+
+STDMETHODIMP CFavBand::CacheItem(long paramC)
+{
+    UNIMPLEMENTED;
+    return E_NOTIMPL;
+}
+
+// *** IDispatch ***
+
+STDMETHODIMP CFavBand::GetTypeInfoCount(UINT *pctinfo)
+{
+    UNIMPLEMENTED;
+    return E_NOTIMPL;
+}
+
+STDMETHODIMP CFavBand::GetTypeInfo(UINT iTInfo, LCID lcid, ITypeInfo **ppTInfo)
+{
+    UNIMPLEMENTED;
+    return E_NOTIMPL;
+}
+
+STDMETHODIMP CFavBand::GetIDsOfNames(
+    REFIID riid,
+    LPOLESTR *rgszNames,
+    UINT cNames,
+    LCID lcid,
+    DISPID *rgDispId)
+{
+    UNIMPLEMENTED;
+    return E_NOTIMPL;
+}
+
+STDMETHODIMP CFavBand::Invoke(
+    DISPID dispIdMember,
+    REFIID riid,
+    LCID lcid,
+    WORD wFlags,
+    DISPPARAMS *pDispParams,
+    VARIANT *pVarResult,
+    EXCEPINFO *pExcepInfo,
+    UINT *puArgErr)
+{
+    switch (dispIdMember)
+    {
+        case DISPID_DOWNLOADCOMPLETE:
+        case DISPID_NAVIGATECOMPLETE2:
+            // FIXME
```

Comment:
What needs fixing here? Can you add a comment?

Reply:
Fixed.